enhancements/oc/day2-add-nodes.md

```diff
@@ -0,0 +1,269 @@
+- As a cluster administrator, I want to validate the nodes to be added before initiating the procedure and be informed of any validation failures
+- As a cluster administrator, I want to be able to monitor the process of adding new nodes
+
+### Goals
+- Allow the user to easily add one or more worker nodes to an existing (disconnected) cluster where MachineSet scaleout is not available or feasible, 
+  with the minimum amount of configuration requested
+- Provide a simpler interface by reusing existing command line tools for cluster administration
+- When adding multiple nodes, a failure in adding a node should not impact the other nodes currently involved in the process
+- Cover at least Baremetal, vSphere, Nutanix and None platform (regardless of the installation method used)
+- For IPI clusters, provide support for the scenario where the user can’t use the hosts BMCs
+
+### Non-Goals
+- Replace MachineSet-based management of hosts for IPI clusters
+- Offer more or finer-grained configuration options for new nodes than those supported by the agent-based installer
+- Generation of the ISO image for adding the nodes into an environment not directly connected to the target cluster
+- Provide an automation for final nodes approval
+
+## Proposal
+A key factor for simplifying the user experience consists in offering a simple enough - but sufficiently flexible -  user interface 
+to accomplish the task of adding the required nodes to a cluster, and to monitor the progress of such activity. For this reason, 
+the proposal consists into enhancing the `oc` cli with two new commands:
+
+- `oc adm add-nodes-image` to generate the ISO for adding the nodes to the cluster
+- `oc adm monitor-add-nodes` to monitor the nodes currently being added to the cluster
+
+There are several reasons for choosing `oc` as a primary user interfaces for these tasks:
+- It’s the official cluster administration tool, so it is expected to be already available in the target cluster environment
+- It already supports the kubeconfig file for interacting with the target cluster (it’s relevant since it contains several pieces of 
+  information required for adding the nodes, as explained in the design section)
+- It already offers a couple of commands to manually approve/deny certificate signing requests (CSRs) when adding new nodes
+
+The usage of `oc` will also help in reducing the amount of information that the user will need to provide, resulting thus in a simplified
+input configuration file.
+
+### add-nodes-image command
+The format of the command will be the following:
+
+```
+oc -–kubeconfig=’’ adm add-nodes-image [-o=’agent-nodes.iso’] <nodes-config.yaml> 
+```
+
+Where:
+-`-o=’agent-nodes.iso`. The name of the ISO image to be generated.
+-`–kubeconfig=’’`. Path to the kubeconfig file (or `KUBECONFIG` if available)
+
+`nodes-config.yaml` is a configuration file, similar to `agent-config.yaml`, used to specify the list of the hosts to be added as worker 
```

Comment:
I think it's important that this file be optional, because having to construct it is a big barrier to entry. The common case will be adding one or more nodes with no network configuration. That should be possible without passing any config.

I think it would also be good to have separate `--network-config=<nmstate.yaml>` and `--root-device=<hint>:<value>` arguments that apply to all nodes booted from the ISO.

And then the nodes-config.yaml would be used only for the case where you are simultaneously adding multiple nodes with static IPs.

Reply:
> I think it's important that this file be optional, because having to construct it is a big barrier to entry. The common case will be adding one or more nodes with no network configuration. That should be possible without passing any config.

These seems reasonable, especially I guess to cover the DHCP scenario, not sure anyhow if the hostnames/macs could be avoided at all. I agree it would be important to make easy the more common scenarios (ie add a node without a specific config)

> I think it would also be good to have separate --network-config=<nmstate.yaml> and --root-device=<hint>:<value> arguments that apply to all nodes booted from the ISO.

Having (some of ) the different pieces captured by the single config file `nodes-config.yaml` seems to me a more complicated interface for the end user, especially in the case of more than one node to be added - and in particular if a `nmstate.yaml` file has to be provided. 

> And then the nodes-config.yaml would be used only for the case where you are simultaneously adding multiple nodes with static IPs.

The main role of the `nodes-config.yaml` is to act as a public API for the end user, since it defines what could be configured or not for a node to be added. It could be also easily expanded in case of need, and stored in a vcs when required.